docs/roadmap.md

```diff
@@ -0,0 +1,169 @@
+* [x] Symbol outline in the sidebar as you type (`textDocument/documentSymbol`).
+* [ ] Goto implementation (`textDocument/implementation`)
+* [ ] Goto type definition (`textDocument/typeDefinition`)
+* [x] Show type of symbol at position (`textDocument/hover`). Requires
+      [semanticdb-scalac](#semanticdb-scalac).
+* [ ] Show type of expression at position (`textDocument/hover`).
+
+## Fast completions
+
+Auto completions are a great way to explore a library interface and save
+keystrokes while writing code. Completions are sensitive to latency since
+suggestions update on every keystroke. Correct and fast completions are
+difficult to support in Scala due to several challenges:
+
+1.  the presentation compiler does not lend itself well to interactive editing
+2.  some completions semantics are are difficult to reproduce outside of the
+    compiler
+    * extension methods depend on implicit conversions
+    * whitebox macros can refine result types which influences completion
+      suggestions
+
+It will take a while until Metals can robustly support completions.
+
+* [x] Scala Presentation Compiler. Faithfully reproduces full language semantics
+      but may be slow and difficult to keep in-sync with the build for
+      long-running interactive sessions.
+  * Disabled by default. Can be enabled with configuration option
+    `scalac.completions.enabled=true`.
+  * Requires [`metalsSetup`](#metalssetup).
+  * Shows parameter list as you type (`textDocument/signatureHelp`).
+* [ ] Auto-import global symbol from classpath.
+* [ ] Auto-expand abstract methods when implementing a trait or abstract class.
+
+## Low CPU and memory usage
+
+Heavy resource usage is one of the most frequent complaints about IDEs. Low CPU
+and memory usage becomes even more important when working in larger codebases.
+
+* [ ] Persistent symbol index: currently Metals stores the navigation index
+      in-memory, which can consume multiple GB of RAM for large projects.
+* [ ] Faster Scala and Java outline indexing. Currently, Metals can index the
+      outlines of Scala sources at ~30-40k loc/s and Java sources at ~450k
+      loc/s.
+* [ ] Throttled background indexing. Currently, all available CPUs are utilized
+      for indexing, which makes the computer slow for a period after running
+      `metalsSetup`.
+* [ ] Faster "open symbol in workspace" (`workspace/symbol`). Currently, uses a
+      naive linear search algorithm that is slow for large projects.
+
+## Solid refactoring
+
+Refactoring makes you more productive while editing code. On demand refactoring
+such as rename or formatting are great to save keystrokes for frequent
+mechanical tasks. Passive refactoring suggestions help you learn new language
+best practices.
+
+* Format (`textDocument/formatting`) with [Scalafmt][].
+  * [x] Whole file on demand (`textDocument/formatting`)
+  * [ ] Selected range (`textDocument/rangeFormatting`)
+  * [x] On save (`textDocument/willSaveWaitUntil`)
```

Comment:
do we actually implement `textDocument/willSaveWaitUntil`?

Reply:
I think @laughedelic added support for it, this is copy-pasted from old roadmap

Reply:
Yes, we have it 👌